sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -516,38 +521,110 @@ class EthService(
     blockTransactionByIndex(nodeView => getBlockIdByTag(nodeView, tag), index)
   }
 
-  private def getPendingState(nodeView: NV, block: AccountBlock, blockId: ModifierId): AccountState = {
-    if (blockId != null) return null
-    nodeView.state.applyModifier(block).getOrElse(throw new RpcException(RpcError.fromCode(
-      RpcCode.InternalError,
-      "Failed to get pending block."
-    )))
+  private def getPendingBlock(nodeView: NV): Option[AccountBlock] = {
+    new AccountForgeMessageBuilder(new MainchainSynchronizer(null), null, networkParams, false)
+      .getPendingBlock(
+        nodeView
+      )
   }
-  private def getPendingStateDbView(nodeView: NV, block: AccountBlock): StateDbAccountStateView = {
-    val pendingState = getPendingState(nodeView, block, null)
-    val stateDbView = pendingState.getStateDbViewFromRoot(block.header.stateRoot)
-    rollbackPendingState(pendingState, idToVersion(block.header.parentId))
-    stateDbView
+
+  private def getPendingBlockInfo(parentId: ModifierId, parentInfo: SidechainBlockInfo): SidechainBlockInfo = {
+    new SidechainBlockInfo(
+      parentInfo.height + 1,
+      parentInfo.score + 1,
+      parentId,
+      System.currentTimeMillis / 1000,
+      ModifierSemanticValidity.Unknown,
+      null,
+      null,
+      WithdrawalEpochUtils.getWithdrawalEpochInfo(
+        0,
+        parentInfo.withdrawalEpochInfo,
+        networkParams
+      ),
```

Comment:
can be replaced with `parentInfo.withdrawalEpochInfo`, since the result of this method will not increase the parent value